cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -197,10 +197,13 @@ public synchronized void delete(final Id.Namespace namespaceId) throws Exception
       // Another reason for not deleting the default namespace is that we do not want to call a delete on the default
       // namespace in the storage provider (Hive, HBase, etc), since we re-use their default namespace.
       if (!Id.Namespace.DEFAULT.equals(namespaceId)) {
-        // Finally delete namespace from MDS
-        nsStore.delete(namespaceId);
-        // Delete namespace in storage providers
-        dsFramework.deleteNamespace(namespaceId);
+        try {
+          // Delete namespace in storage providers
+          dsFramework.deleteNamespace(namespaceId);
+        } finally {
+          // Finally delete namespace from MDS
+          nsStore.delete(namespaceId);
```

Comment:
I think this is fine. There are problems in the delete/create flow today, but I don't think this PR increases/decreases them.